### ISSUE TITLE
DynamoDB crc32 mismatch

### ISSUE THREAD

**Issue by mikaelmattsson** [localstack/localstack#429]
`crc32 mismatch. Expected 90958358, found 2185086916.`

I get this error when fetching a large dataset.

The aws-php-sdk is throwing the error here:

```
        if ($expected = $response->getHeaderLine('x-amz-crc32')) {
            $hash = hexdec(Psr7\hash($response->getBody(), 'crc32b'));
            if ($expected != $hash) {
                throw new AwsException(
                    "crc32 mismatch. Expected {$expected}, found {$hash}.",
                    $command,
                    [
                        'code'             => 'ClientChecksumMismatch',
                        'connection_error' => true,
                        'response'         => $response
                    ]
                );
            }
        }
```

I suspect there is a problem with localstack gzipping the data but not changing the x-amz-crc32 header to the correct value.

I also sometimes get this error:

`AWS HTTP error: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding`

**Comment by whummer** [comment]
Thanks for reporting @mikaelmattsson . Any chance you could create a small project on Github with a self-contained test case to reproduce this error? Thanks

**Comment by mikaelmattsson** [comment]
https://github.com/mikaelmattsson/localstack-issue-429

I was unable to test this because for some reason my localstack has stopped responding. This should work however. Will try again later on a different computer.

**Comment by mikaelmattsson** [comment]
Confirmed on a different computer:

```Running query. Expecting exception.PHP Fatal error:  Uncaught exception 'Aws\DynamoDb\Exception\DynamoDbException' with message 'Error executing "Query" on "http://localstack:4569"; AWS HTTP error: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) Unable to parse error information from response - Error parsing JSON: Syntax error'

GuzzleHttp\Exception\RequestException: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) in /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:187
Stack trace:
#0 /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(150): GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle), Array)
#1 /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(103): GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlMultiHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
#2 /app/vendor/guzz in /app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 192

Fatal error: Uncaught exception 'Aws\DynamoDb\Exception\DynamoDbException' with message 'Error executing "Query" on "http://localstack:4569"; AWS HTTP error: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) Unable to parse error information from response - Error parsing JSON: Syntax error'

GuzzleHttp\Exception\RequestException: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding (see http://curl.haxx.se/libcurl/c/libcurl-errors.html) in /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:187
Stack trace:
#0 /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(150): GuzzleHttp\Handler\CurlFactory::createRejection(Object(GuzzleHttp\Handler\EasyHandle), Array)
#1 /app/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(103): GuzzleHttp\Handler\CurlFactory::finishError(Object(GuzzleHttp\Handler\CurlMultiHandler), Object(GuzzleHttp\Handler\EasyHandle), Object(GuzzleHttp\Handler\CurlFactory))
#2 /app/vendor/guzz in /app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 192
```

**Comment by whummer** [comment]
Thanks for reporting @mikaelmattsson . This should be fixed in #482. Can you please give it a try with the latest version of the Docker image (tag `latest`, currently being built by TravisCI). Closing this issue now, as I have been able to run your test with the new version. Please report back here if the problem persists. Thanks
